### PR TITLE
Add support for the x64 platform in wixproj templates

### PR DESCRIPTION
**Pull request by lb90**
Just a proposal...

Currently the wixproj templates only support the x86 platform. It's easy to add support for the x64 platform with some manual editing, but having that ready out of the box would be great.

I'm not sure if something can be done better

### CODE REVIEW

File: src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj [hunk 1]
```diff
@@ -75,6 +108,27 @@
...
+			<MinimalRebuild>true</MinimalRebuild>
+			<BasicRuntimeChecks>EnableFastChecks</BasicRuntimeChecks>
+			<RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
+			<PrecompiledHeader>Use</PrecompiledHeader>
+			<WarningLevel>Level3</WarningLevel>
+			<DebugInformationFormat>EditAndContinue</DebugInformationFormat>
+		</ClCompile>
+		<Link>
+			<AdditionalDependencies>msi.lib;dutil.lib;wcautil.lib;Version.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalLibraryDirectories>$(WIX)sdk\$(WixPlatformToolset)\lib\x86;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
Comment (rseanhall) [inline src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj]: This needs to be x64.

File: src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj [hunk 2]
```diff
@@ -98,6 +152,29 @@
...
+			<PreprocessorDefinitions>WIN32;NDEBUG;_WINDOWS;_USRDLL;CUSTOMACTIONTEST_EXPORTS;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+			<RuntimeLibrary>MultiThreaded</RuntimeLibrary>
+			<FunctionLevelLinking>true</FunctionLevelLinking>
+			<PrecompiledHeader>Use</PrecompiledHeader>
+			<WarningLevel>Level3</WarningLevel>
+			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
+		</ClCompile>
+		<Link>
+			<AdditionalDependencies>msi.lib;dutil.lib;wcautil.lib;Version.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalLibraryDirectories>$(WIX)sdk\$(WixPlatformToolset)\lib\x86;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
Comment (rseanhall) [inline src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj]: This needs to be x64.

**Review by rseanhall — CHANGES_REQUESTED** [review]
Thanks. This seems to be the right way to do this.

File: src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj (hunk 1 above)
Comment (lb90) [reply src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj]: Good catch! Fixed.

File: src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj (hunk 2 above)
Comment (lb90) [reply src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj]: Fixed

**Comment by lb90** [comment]
Added more project templates, should be complete by now.

Also added a .wxi include file to the MSI Project template. This is to help with installing in either ProgramFilesFolder or ProgramFiles64Folder, depending on the active Platform configuration. It contains a simple 'PlatformProgramFilesFolder' define. That include file is named 'Platform.wxi', but perhaps we could change the name to 'TargetArch.wxi' or 'TargetPlatform.wxi', any suggestion about naming is welcome.

Thanks!

File: src/Templates/v3/Projects/WixBundleProject/BundleProject.wixproj [hunk 3]
```diff
@@ -10,13 +10,22 @@
...
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+		<DefineConstants>Debug</DefineConstants>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
```
Comment (rseanhall) [inline src/Templates/v3/Projects/WixBundleProject/BundleProject.wixproj:28]: Bundles are always x86 in v3, so I don't think we should add an x64 configuration.

File: src/Templates/v3/Projects/WixLibrary/setuplibrary.wixproj [hunk 4]
```diff
@@ -10,13 +10,22 @@
...
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+		<DefineConstants>Debug</DefineConstants>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
```
Comment (rseanhall) [inline src/Templates/v3/Projects/WixLibrary/setuplibrary.wixproj:28]: I don't think library projects should have multiple platforms. Best practice is to build one .wixlib with all the architectures, with different `Id`s based on the target architecture.

File: src/Templates/v3/Projects/WixProject/Product.wxs [hunk 5]
```diff
@@ -13,7 +13,7 @@
 
 	<Fragment>
 		<Directory Id="TARGETDIR" Name="SourceDir">
-			<Directory Id="ProgramFilesFolder">
+			<Directory Id="$(var.PlatformProgramFilesFolder)">
```
Comment (rseanhall) [inline src/Templates/v3/Projects/WixProject/Product.wxs:16]: I don't think we should do this. You shouldn't build the project as x86 and as x64 and then distribute them both. You would need to vary other things as well, such as the UpgradeCode.

File: src/Templates/v4/Projects/WixBundleProject/BundleProject.wixproj [hunk 6]
```diff
@@ -11,13 +11,22 @@
 		<WixTargetsPath Condition=" '$(WixTargetsPath)' == '' ">$(MSBuildExtensionsPath)\WiX Toolset\v4\wix.targets</WixTargetsPath>
 	</PropertyGroup>
 	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|x86' ">
-		<OutputPath>bin\$(Configuration)\</OutputPath>
-		<IntermediateOutputPath>obj\$(Configuration)\</IntermediateOutputPath>
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
```
Comment (rseanhall) [inline src/Templates/v4/Projects/WixBundleProject/BundleProject.wixproj:15]: I don't want to touch any of the v4 templates. The projects will look completely different.

**Comment by AraHaan** [comment]
What about arm64 to the templates as well?

**Comment by robmen** [comment]
Interesting proposal. Feel free to send a completed solution in the future. For now, closing the stale PR.

**Pull request closed without merging**